### PR TITLE
[10.0][MIG] procurement_auto_create_group

### PR DESCRIPTION
**Pull request by LoisRForgeFlow**
Migration to v10

Procurement Auto Create Group
=============================

This module allows the system to propose automatically new procurement groups
in procurement orders.

This capability is important when you want to make sure that all the stock
moves resulting from this procurement will never be mixed with moves from
other groups in stock transfers.

The stock transfers resulting from the execution of these procurements will
only contain stock moves created from that procurement.

Configuration
=============

#. Go to *Inventory / Configuration / Settings* and check the option 'Advanced
   routing of products using rules' and press the 'Apply' button.
#. Activate the developer mode.
#. Go to *Inventory / Configuration / Routes / Routes* and check the option
   'Auto-create Procurement Group' to the pull rules where you want the
   procurement groups to be automatically proposed.

Usage
=====

#. Go to *Inventory / Reports / Procurement Exceptions*.
#. Create a new procurement order and make sure that it determines a pull rule
   with the option 'Auto-create Procurement Group' set.
#. When you save the procurement order, a procurement group with format
   'PG/000001' will be created.

### CODE REVIEW

File: procurement_auto_create_group/models/procurement.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    def _prepare_auto_procurement_group_data(self):
+        name = self.env['ir.sequence'].next_by_code(
+            'procurement.group') or False
+        if not name:
+            raise UserError(_('No sequence defined for procurement group'))
+        return {
+            'name': name
+        }
+
+    @api.model
```
Comment (rousseldenis) [inline procurement_auto_create_group/models/procurement.py]: api.multi

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Minor comment. LGTM

**Comment by LoisRForgeFlow** [comment]
@rousseldenis Thanks, fixed!

**Review by rousseldenis — APPROVED** [review]
LGTM. Code review

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by lreficent** [comment]
rebase and squased

**Comment by LoisRForgeFlow** [comment]
rebased and squased

**Pull request merged**